test/unit/module/core/logical_test.hpp

```diff
@@ -0,0 +1,191 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+**/
+//==================================================================================================
+#pragma once
+
+#include "test.hpp"
+
+#include <eve/module/core.hpp>
+
+using namespace eve;
+
+template <typename A, typename B>
+static constexpr auto find_priority_type()
+{
+  using ea_t = eve::as_arithmetic_t<eve::element_type_t<A>>;
+  using eb_t = eve::as_arithmetic_t<eve::element_type_t<B>>;
+
+  if constexpr (std::same_as<ea_t, eb_t>)      return A{};
+  // smallest type
+  else if constexpr (sizeof(ea_t) != sizeof(eb_t))
+  {
+    if constexpr (sizeof(ea_t) < sizeof(eb_t)) return A{};
+    else                                       return B{};
+  }
+  // unsigned first
+  else if constexpr (eve::signed_value<ea_t> != eve::signed_value<eb_t>)
+  {
+    if constexpr (eve::signed_value<ea_t>)     return B{};
+    else                                       return A{};
+  }
+  // integral first
+  else if constexpr (eve::integral_value<ea_t> != eve::integral_value<eb_t>)
+  {
+    if constexpr (eve::integral_value<ea_t>)   return A{};
+    else                                       return B{};
+  }
+  else                                         return A{};
+}
+
+template<typename T, typename U>
+using priority_t = eve::as_logical_t<decltype(find_priority_type<T, U>())>;
```

Comment:
As I mentioned - this repeats a bit common_logical type, so the value of such testing is limited.